File: Seminar5/HomeWork/PresenterStartRation.java
package HomeWork;

import java.util.Scanner;
import java.util.logging.*;

import HomeWork.Controller.PresenterRation;
import HomeWork.Model.Rational.DivRational;
import HomeWork.Model.Rational.MultRational;
import HomeWork.Model.Rational.SubModelRational;
import HomeWork.Model.Rational.SumRational;
import HomeWork.View.Menu;
import HomeWork.View.ViewRational;

public class PresenterStartRation {

    PresenterRation presenterRational;
    Menu menu;
    Scanner scanner;

    public PresenterStartRation(Menu menu, Scanner scanner) {
        this.menu = menu;
        this.scanner = scanner;
    }

    public void startRation() {
        Logger logger = Logger.getGlobal();
        switch (menu.selectFunction()) {
            case 1:
                System.out.println("Какие числа будем суммировать?");
                PresenterRation p1 = new PresenterRation(new SumRational(), new ViewRational());
                p1.buttonClick();
                logger.info("Вычисление прошло успешно");
                break;
            case 2:
                System.out.println("Какие числа будем вычитать?");
                PresenterRation p2 = new PresenterRation(new SubModelRational(), new ViewRational());
                p2.buttonClick();
                logger.info("Вычисление прошло успешно");

                break;
            case 3:
                System.out.println("Какие числа будем умножать?");
                PresenterRation p3 = new PresenterRation(new MultRational(), new ViewRational());
                p3.buttonClick();
                logger.info("Вычисление прошло успешно");

                break;
            case 4:
                System.out.println("Какие числа будем делить?");
                PresenterRation p4 = new PresenterRation(new DivRational(), new ViewRational());
                p4.buttonClick();
                logger.info("Вычисление прошло успешно");

                break;
            default:
                System.out.println("Неверный ввод");
                logger.info("Ошибка");
                break;
        }

    }

}

File: FinalTaskCalculator/PresenterStartRation.cs
using System;
using static Controller.PresenterRation;
using static ModelRational.DivRational;
using static ModelRational.MultRational;
using static ModelRational.SubModelRational;
using static ModelRational.SumRational;
using static View.Menu;
using static View.ViewRational;

public class PresenterStartRation
{
    View.Menu menu = null!;

    public PresenterStartRation(View.Menu menu)
    {
        this.menu = menu;
    }

    public void StartRation()
    {
        switch (menu.SelectFunction())
        {
            case 1:
                Console.WriteLine("Какие числа будем суммировать?");
                Controller.PresenterRation p1 = new Controller.PresenterRation(new ModelRational.SumRational(), new View.ViewRational());
                p1.ButtonClick();
                Console.WriteLine("Вычисление прошло успешно");
                break;
            case 2:
                Console.WriteLine("Какие числа будем вычитать?");
                Controller.PresenterRation p2 = new Controller.PresenterRation(new ModelRational.SubModelRational(), new View.ViewRational());
                p2.ButtonClick();
                Console.WriteLine("Вычисление прошло успешно");

                break;
            case 3:
                Console.WriteLine("Какие числа будем умножать?");
                Controller.PresenterRation p3 = new Controller.PresenterRation(new ModelRational.MultRational(), new View.ViewRational());
                p3.ButtonClick();
                Console.WriteLine("Вычисление прошло успешно");

                break;
            case 4:
                Console.WriteLine("Какие числа будем делить?");
                Controller.PresenterRation p4 = new Controller.PresenterRation(new ModelRational.DivRational(), new View.ViewRational());
                p4.ButtonClick();
                Console.WriteLine("Вычисление прошло успешно");

                break;
            default:
                Console.WriteLine("Неверный ввод");
                Console.WriteLine("Ошибка");
                break;
        }

    }

}